File: LO52_2019_FGurlsDev/Part 4/Tp4Jni/app/src/main/java/fr/utbm/fgurlsdev/tp4jni/MainActivity.java
package fr.utbm.fgurlsdev.tp4jni;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    private EditText editText;
    private TextView textShow;
    private Button btnRead;
    private Button btnWrite;
    private Button btnUp;
    private Button btndown;
    private Button btnLeft;
    private Button btnRight;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Example of a call to a native method
        TextView tv = findViewById(R.id.sample_text);
        tv.setText(NDKTools.sayHi());

        initView();
    }

    @Override
    protected void onResume() {
        super.onResume();

        btnRead.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String editContent = editText.getText().toString();
                if(!Outils.isCorrectContent(editContent)) Toast.makeText(MainActivity.this,"Please enter a number between 0 and 10!", Toast.LENGTH_SHORT).show();
                else textShow.setText(NDKTools.readBtn(editContent));
            }
        });

        btnWrite.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String editContent = editText.getText().toString();
                if(!Outils.isCorrectContent(editContent)) Toast.makeText(MainActivity.this,"Please enter a number between 0 and 10!", Toast.LENGTH_SHORT).show();
                else textShow.setText(NDKTools.writeBtn(editContent));
            }
        });

        btnUp.setOnClickListener(new DirectionBtnListener(textShow));
        btndown.setOnClickListener(new DirectionBtnListener(textShow));
        btnLeft.setOnClickListener(new DirectionBtnListener(textShow));
        btnRight.setOnClickListener(new DirectionBtnListener(textShow));
    }

    private void initView(){
        editText = findViewById(R.id.editText);
        textShow = findViewById(R.id.show_text);
        btnRead = findViewById(R.id.btn_read);
        btnWrite = findViewById(R.id.btn_write);
        btnUp = findViewById(R.id.btn_up);
        btndown = findViewById(R.id.btn_down);
        btnLeft = findViewById(R.id.btn_left);
        btnRight = findViewById(R.id.btn_right);
    }
}

File: LO52_2019_Equal/TP4/app/src/main/java/com/startup42/tp4/MainActivity.kt
package com.startup42.tp4

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_main.*



class MainActivity : AppCompatActivity() {

    private external fun readJNI(): Int
    private external fun writeJNI(): Int
    private external fun directionJNI(direction: String): String

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Example of a call to a native method
        readButton.setOnClickListener {
            val a = readJNI()
            textView.text = "Read : ${a}"
        }

        writeButton.setOnClickListener {
            val a = writeJNI()
            textView.text = "Write : ${a}"
        }

        upButton.setOnClickListener {
            textView.text = directionJNI(upButton.text.toString())
        }

        downButton.setOnClickListener {
            textView.text = directionJNI(downButton.text.toString())
        }

        leftButton.setOnClickListener {
            textView.text = directionJNI(leftButton.text.toString())
        }

        rightButton.setOnClickListener {
            textView.text = directionJNI(rightButton.text.toString())
        }
    }

    /**
     * A native method that is implemented by the 'native-lib' native library,
     * which is packaged with this application.
     */
    external fun stringFromJNI(): String

    companion object {

        // Used to load the 'native-lib' library on application startup.
        init {
            System.loadLibrary("native-lib")
        }
    }
}
